cpp/arcticdb/storage/storage.hpp

```diff
@@ -158,8 +158,16 @@ class Storage {
         return do_key_exists(key);
     }
 
+    bool key_exists(KeyType key_type, const IterateTypePredicate& predicate) {
```

Comment:
Is this potentially a source of confusion given the radically different performance characteristics of the two methods? One is constant and the other is worst-case linear in the number of keys of that type - a difference of less than a second vs potentially several hours. I think I'd prefer if the scanning version was more up-front about what it's doing - scan_for_matching_key or something like that